### PR TITLE
feat(subscription): add common JSON-LD and HTTP behaviours

### PR DESCRIPTION
**Pull request by bobeal**

### CODE REVIEW

**Comment by bobeal** [comment]
@gpoujol sorry, it's a huge PR ... take your time to review, some things have most probably been missed.

File: subscription-service/src/main/kotlin/com/egm/stellio/subscription/utils/DBConversionUtils.kt [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package com.egm.stellio.subscription.utils
```
Comment (gpoujol) [inline subscription-service/src/main/kotlin/com/egm/stellio/subscription/utils/DBConversionUtils.kt:1]: There are the same file in `search_service`. Maybe put them in `shared` folder ?

File: subscription-service/src/main/kotlin/com/egm/stellio/subscription/utils/DBQueryUtils.kt [hunk 2]
```diff
@@ -1,18 +1,47 @@
 package com.egm.stellio.subscription.utils
```
Comment (gpoujol) [inline subscription-service/src/main/kotlin/com/egm/stellio/subscription/utils/DBQueryUtils.kt:1]: Same as above ?

File: subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/EntityEventListenerServiceTests.kt [hunk 3]
```diff
@@ -25,46 +25,42 @@ class EntityEventListenerServiceTests {
...
+        } returns listOf(
+            Triple(mockedSubscription, mockedNotification, true),
+            Triple(mockedSubscription, mockedNotification, false)
+        ).right()
 
-        entityEventListenerService.processMessage(updateEvent)
+        entityEventListenerService.dispatchEntityEvent(updateEvent)
 
-        verify { notificationService.notifyMatchingSubscribers(any(), any(), setOf("name")) }
+        coVerify { notificationService.notifyMatchingSubscribers(any(), any(), setOf("https://schema.org/name")) }
```
Comment (gpoujol) [inline subscription-service/src/test/kotlin/com/egm/stellio/subscription/service/EntityEventListenerServiceTests.kt:63]: Maybe replace by `NGSILD_NAME_PROPERTY` ?

**Review by gpoujol — APPROVED** [review]

**Review by gpoujol — APPROVED** [review]

**Pull request merged**